### PR TITLE
[1LP][RFR] Fixed TypeError while collecting hosts

### PR DESCRIPTION
**Pull request by ganeshhubale**
It fixes 
1. 'type error' 
2. 'HostCollection object is not iterable error'

{{ pytest: cfme/tests/infrastructure/test_tenant_quota.py::test_vm_migration_after_assigning_tenant_quota --use-provider complete --long-running }}

### CODE REVIEW

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 1]
```diff
@@ -107,10 +107,10 @@ def small_vm(provider, small_template_modscope):
 @pytest.fixture()
 def check_hosts(small_vm, provider):
     """Fixture to return host"""
-    if len(provider.hosts) != 1:
+    if provider.hosts != 1:
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_tenant_quota.py]: Required: this will be always true as provider.hosts is hosts collection object
you need to use provider.hosts.all() this will return you a list all available hosts and len(provider.hosts.all()) will do the trick

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 2]
```diff
@@ -253,7 +253,7 @@ def test_setting_child_quota_more_than_parent(tenants_setup, parent_quota, child
 
 
 @pytest.mark.long_running
-@pytest.mark.provider([VMwareProvider, RHEVMProvider], override=True, scope="module",
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_tenant_quota.py]: Question: Why did you remove RHEVMProvider provider?

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 3]
```diff
@@ -253,7 +253,7 @@ def test_setting_child_quota_more_than_parent(tenants_setup, parent_quota, child
...
+@pytest.mark.provider([VMwareProvider], override=True, scope="module",
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_tenant_quota.py]: Request: we don't use scope="module" for providers, please remove

File: cfme/tests/infrastructure/test_tenant_quota.py (hunk 2 above)
Comment (ganeshhubale) [reply cfme/tests/infrastructure/test_tenant_quota.py]: RHEVMProvider provider  is not providing multiple hosts. Because of this, vm migration gets failed.

File: cfme/tests/infrastructure/test_tenant_quota.py (hunk 1 above)
Comment (ganeshhubale) [reply cfme/tests/infrastructure/test_tenant_quota.py]: Thanks

File: cfme/tests/infrastructure/test_tenant_quota.py (hunk 3 above)
Comment (ganeshhubale) [reply cfme/tests/infrastructure/test_tenant_quota.py]: @rlbabyuk 
To fix this issue. If I removed **scope='module'** and changed it to **scope='function'** then I need some changes in following files too:
```
1. cfme/tests/infrastructure/test_tenant_quota.py:95:  def small_vm(provider, small_template_modscope) 
2. cfme/fixtures/provider.py:357:  def small_template_modscope(provider) 
```
In above files: need to change **scope='function'**. 
If it is not affecting other tests then we can go ahead with these changes.
please let me know if you have any suggestions for me.

Comment (rlbabyuk) [reply cfme/tests/infrastructure/test_tenant_quota.py]: @ganeshhubale I think in this case we can leave provider scope as module, not to change other fixtures
